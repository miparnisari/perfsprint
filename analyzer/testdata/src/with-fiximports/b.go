package a

import (
	"strings"
)

func loopconcat2() {
	// already strings
	s := ""
	for i := 0; i < 10; i++ {
		s += "toto" // want "concat-loop: string concatenation in a loop"
	}
	println("Hello, World!", strings.Fields(s))
}
