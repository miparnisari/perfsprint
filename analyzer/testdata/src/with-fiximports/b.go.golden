package a

import (
	"strings"
)

func loopconcat2() {
	// already strings
	s := ""
	var sSb10 strings.Builder
	for i := 0; i < 10; i++ {
		sSb10.WriteString("toto") // want "concat-loop: string concatenation in a loop"
	}
	s += sSb10.String()
	println("Hello, World!", strings.Fields(s))
}
